docs/release-info/release-notes.md

```diff
@@ -24,6 +24,26 @@ title: "Release notes"
 
 <!--Replace {{DRUIDVERSION}} with the correct Druid version.-->
 
+## Druid 29.0.1
+
+Apache Druid 29.0.1 is a patch release that fixes some issues in the Druid 29.0.0 release.
+
+### Bug fixes
+
+* Added type verification for INSERT and REPLACE to validate that strings and string arrays aren't mixed [#15920](https://github.com/apache/druid/pull/15920)
+* [Concurrent replace](https://druid.apache.org/docs/latest/ingestion/concurrent-append-replace) now allows pending Peon segments to be upgraded using the Supervisor [#15995](https://github.com/apache/druid/pull/15995)
+* Changed the `targetDataSource` attribute to return a string containing the name of the datasource [#16004](https://github.com/apache/druid/pull/16004)
```

Comment:
We should also mention that This reverts the accidental change as capture in: https://github.com/apache/druid/pull/16031
cc @adarshsanjeev